ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/validation/CustomValidators.java

```diff
@@ -0,0 +1,219 @@
+/*
+* Copyright 2019 Confluent Inc.
+*
+* Licensed under the Confluent Community License; you may not use this file
+* except in compliance with the License.  You may obtain a copy of the License at
+*
+* http://www.confluent.io/confluent-community-license
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+* WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations under the License.
+*/
+
+package io.confluent.ksql.rest.server.validation;
+
+import com.google.common.collect.ImmutableMap;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.confluent.ksql.KsqlExecutionContext;
+import io.confluent.ksql.config.KsqlConfigResolver;
+import io.confluent.ksql.function.FunctionRegistry;
+import io.confluent.ksql.metastore.KsqlTopic;
+import io.confluent.ksql.metastore.StructuredDataSource;
+import io.confluent.ksql.parser.KsqlParser.PreparedStatement;
+import io.confluent.ksql.parser.tree.DescribeFunction;
+import io.confluent.ksql.parser.tree.Explain;
+import io.confluent.ksql.parser.tree.PrintTopic;
+import io.confluent.ksql.parser.tree.Query;
+import io.confluent.ksql.parser.tree.SetProperty;
+import io.confluent.ksql.parser.tree.ShowColumns;
+import io.confluent.ksql.parser.tree.Statement;
+import io.confluent.ksql.parser.tree.TerminateQuery;
+import io.confluent.ksql.parser.tree.UnsetProperty;
+import io.confluent.ksql.query.QueryId;
+import io.confluent.ksql.rest.server.execution.ExplainExecutor;
+import io.confluent.ksql.rest.server.resources.Errors;
+import io.confluent.ksql.rest.server.resources.KsqlRestException;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlException;
+import io.confluent.ksql.util.KsqlStatementException;
+import java.util.Map;
+
+/**
+ * A set of {@code StatementValidator}s which are used to validate non-executable
+ * statements. Each handles a corresponding {@code Class<? extends Statement>} and
+ * is assumed that the {@code PreparedStatement} that is passed in matches the
+ * expected class.
+ */
+public enum CustomValidators implements StatementValidator {
```

Comment:
Likewise, swap this from an enum to individual types.

Also, there are some types where the validation _is the same as the execution_. (I'll call them out below).